var/spack/repos/builtin/packages/gtkorvo-atl/package.py

```diff
@@ -35,10 +35,14 @@ class GtkorvoAtl(CMakePackage):
 
     version('develop', git='https://github.com/GTkorvo/atl.git',
             branch='master')
+    version('2.2', 'f0e3581e4b4c6943bf4b203685630564')
     version('2.1', 'b2324ff041bccba127330a0e1b241978')
 
     depends_on('gtkorvo-cercs-env')
 
     def cmake_args(self):
-        args = ["-DENABLE_TESTING=0", "-DENABLE_BUILD_STATIC=STATIC"]
+        if self.spec.satisfies('@2.2:'):
+            args = ["-DENABLE_TESTING=0", "-DBUILD_SHARED_LIBS=OFF"]
+        else:
+            args = ["-DENABLE_TESTING=0", "-DENABLE_BUILD_STATIC=STATIC"]
```

Comment:
To remove the duplication, can you do:
```python
args = ['-DENABLE_TESTING=0']

if self.spec.satisfies('@2.2:'):
    args.append('-DBUILD_SHARED_LIBS=OFF')
else:
    args.append('-DENABLE_BUILD_STATIC=STATIC')
```
Also, is `-DENABLE_TESTING` needed for running `make test`? If so, you could do:
```python
if self.run_tests:
    args.append('-DENABLE_TESTING=1')
else:
    args.append('-DENABLE_TESTING=0')
```
`self.run_tests` is a Boolean that says whether `spack install` was run with the `--test` flag. If it was, Spack runs `make test` automatically.

Reply:
Yes, this would likely be a good change...

Reply:
`spack install --test=all libevpath` fails during installation of libffs with 
```
92% tests passed, 6 tests failed out of 74

Total Test time (real) =   8.86 sec

The following tests FAILED:
         22 - context_test (Failed)
         23 - context_test2 (SEGFAULT)
         24 - fortran_test (SEGFAULT)
         29 - write_context_output (Failed)
         30 - context_output_unique (Failed)
         52 - CTOUT_context_test_output.bin (Failed)
Errors while running CTest
Makefile:130: recipe for target 'test' failed
make: *** [test] Error 8
```

This is on a local Debian machine.
@eisenhauer Is this known? Is it ok to proceed with this change `if self.run_tests`?

Reply:
No, not expected...  Let me see if I can duplicate on my Ubuntu VM, just because that's the quickest.  (But generally we're doing nightly regression tests on this stuff, so it'd be weird to have a problem on debian or in spack that didn't show up, say, on redhat.  Really this stuff is pretty OS agnostic, with architecture differences being a far bigger problem.)  BTW, I am having to switch machines today to send my MacBook in for service.  I've transfered my VM to a backup, so I'm hoping for little impact, but can't make promises.

Reply:
Also, how does `-DENABLE_TESTING=0` affect the tests?
I set `-DENABLE_TESTING=0` for all cases and ran `spack install --tests=all libevpath`. 
```
if self.run_tests:
    args.append('-DENABLE_TESTING=0')
else:
    args.append('-DENABLE_TESTING=0')
```

libffs fails at the testing phase even if -DENABLE_TESTING=0.